lib/seed/reservation_generator_helper.rb

```diff
@@ -0,0 +1,95 @@
+module ReservationGeneratorHelper
+  include ActiveSupport::Testing::TimeHelpers
+
+  def gen_res(random = false)
+    r = Reservation.new(status: 'reserved', reserver_id: User.all.sample.id,
+                        equipment_model: EquipmentModel.all.sample,
+                        notes: FFaker::HipsterIpsum.paragraph(2),
+                        start_date: Time.zone.today)
+    len = r.equipment_model.maximum_checkout_length
+    len -= random ? rand(len - 1) : 1
+    r.due_date = r.start_date + len.days
+    r
+  end
+
+  def make_checked_out(res, _ = false)
+    res.status = 'checked_out'
+    res.checked_out = res.start_date
+    res.equipment_item = res.equipment_model.equipment_items.all.sample
+    res.checkout_handler_id = User.where('role = ? OR role = ? OR role = ?',
+                                         'checkout', 'admin',
+                                         'superuser').all.sample.id
+  end
+
+  def make_returned(res, random = false)
+    make_past(res, random)
+    make_checked_out res
+    r_date = if res.due_date > Time.zone.today
+               random ? rand(res.start_date..Time.zone.today) : Time.zone.today
+             else
+               random ? rand(res.start_date..res.due_date) : res.due_date
```

Comment:
I feel like the handling of `random` could be done in a cleaner way, but I'm not sure how yet